src/lib/getElementType.js

```diff
@@ -0,0 +1,41 @@
+import _ from 'lodash'
+
+/**
+ * Returns a createElement() type based on the props of the Component.
+ * Useful for calculating what type a component should render as.
+ *
+ * @param {function} Component A function or ReactClass.
+ * @param {object} props A ReactElement props object
+ * @param {object} [typeMap] Maps props (keys) to element types (values).
+ * @returns {string|function} A ReactElement type
+ */
+function getElementType(Component, props, typeMap) {
+  const { defaultProps = {} } = Component
+
+  // ----------------------------------------
+  // user defined element type
+
+  if (props.as && props.as !== defaultProps.as) return props.as
```

Comment:
Let me go back and figure out why I did this, could be cruft.  Used to have `defaultAs` prop here.